common/lib/dependabot/dependency_file.rb

```diff
@@ -40,6 +40,9 @@ class DependencyFile
     sig { returns(T.nilable(String)) }
     attr_accessor :mode
 
+    sig { returns(T::Set[Dependabot::DependencyFile]) }
+    attr_accessor :dependencies
```

Comment:
So, thinking this out - the consequences of this addition is that each ecosystem must now attach its dependency list to the file within it's `file_parser` as we've done for bundler in order to be compatible with Dependabot-driven analysis.

I don't see any major downsides to this, the long tail will be modifying the base file parser to make this method obligatory once we've backfilled the behaviour to extant ecosystems which seems fine.